backend/multimodalexplorer/main.py

```diff
@@ -29,4 +30,12 @@ async def root():
     return {"message": "Welcome to MultiModalExplorer."}
 
 
+def main():
```

Comment:
https://github.com/gwenzek/func_argparse is a good lib to parse cli arguments in python so you can pass some parameters here.

Reply:
I don't think that preprocessing should live in the server main. I suggest that you have:
1. a preprocessing script that takes a (set of) dataset as input and creates file in a pre-determined folder structure
2. a server script that can load the preprocessed data. This could be static at the beginning, where it the server startup script takes as a parameter the name of the folder where to find the preprocessed file for a (set of) datasets we want to look at.